### PR TITLE
feat(zeebe): identity token validation in gateway

### PR DESCRIPTION
**Pull request by megglos**
## What is the purpose of the change

This adds the documentation for the recently added OAuth token validation support to the Zeebe Gateway. See https://github.com/camunda/product-hub/issues/120.

## Are there related marketing activities

No

## When should this change go live?

Ideally with the 8.2.0 release, also fine if shortly after.

## PR Checklist

- [ ] My changes apply to an already released version, and I have added them to the relevant `/versioned_docs` directory, or they are not for an already released version.
- [x] My changes apply to future versions, and I have added them to the main `/docs` directory, or they are not for future versions.
- [ ] My changes require an [Engineering review](https://github.com/camunda/camunda-platform-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned an engineering manager or tech lead as a reviewer, or my changes do not require an Engineering review.
- [ ] My changes require a [technical writer review](https://github.com/camunda/camunda-platform-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned @christinaausley as a reviewer, or my changes do not require a technical writer review.

### CODE REVIEW

**Comment by megglos** [comment]
@christinaausley @akeller  the link checks failed with `connection failed` is that a CI hiccup? 😅

**Comment by akeller** [comment]
@megglos it happens when people work on their holiday 😝 

It's a CI hiccup. I'll rerun it for you.

**Comment by akeller** [comment]
1 link error detected, I'll get that fixed now.

File: docs/self-managed/zeebe-deployment/security/client-authorization.md [hunk 1]
```diff
@@ -226,12 +159,100 @@ func main() {
...
+
+```go
+package main
+
+import (
+    "context"
+    "fmt"
+    "google.golang.org/grpc/status"
+    "google.golang.org/grpc/codes"
+    "github.com/camunda-cloud/zeebe/clients/go/pkg/zbc"
```
Comment (akeller) [inline docs/self-managed/zeebe-deployment/security/client-authorization.md:225]: Is this the correct import statement? Or should it be `/camunda/`?

Comment (megglos) [reply docs/self-managed/zeebe-deployment/security/client-authorization.md:225]: that appears outdated indeed, so we may also fix that in the older versions (I just moved this section without modifying it)

Comment (akeller) [reply docs/self-managed/zeebe-deployment/security/client-authorization.md:225]: I'll create an issue and we can handle that separately. Thanks!

**Comment by megglos** [comment]
> When do we capitalize and lowercase gateway/Zeebe Gateway? How do we refer to this in the UI and how do we want to refer to this in the docs? Do you have any thoughts here?

I'm no aware of any references to it from the UI, e.g. the Modeler refers to it as `cluster endpoint` 
image

Just from my personal opinion I would refer to it as capitalised `Zeebe Gateway` as it's a distinct component of the stack, thus also refer to brokers as `Zeebe Broker`

**Comment by christinaausley** [comment]
@megglos Cool 😄 I launched [an issue](https://github.com/camunda/developer-experience/issues/58) in the backlog to take a closer look and make sure terms/capitalization is consistent across docs.

**Pull request merged**